### ISSUE TITLE
Python example in README fails

### ISSUE THREAD

**Issue by michaelmior** [alipay/fury#1217]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/alipay/issues) and found no similar issues.

### Version

Fury: 0.4.0
OS: Ubuntu 20.04
Python: 3.12.0

### Component(s)

Python

### Minimal reproduce step

I ran the example below copied directly from the README.

```
from typing import Dict
import pyfury

class SomeClass:
    f1: "SomeClass"
    f2: Dict[str, str]
    f3: Dict[str, str]

fury = pyfury.Fury(ref_tracking=True)
fury.register_class(SomeClass, "example.SomeClass")
obj = SomeClass()
obj.f2 = {"k1": "v1", "k2": "v2"}
obj.f1, obj.f3 = obj, obj.f2
data = fury.serialize(obj)
# bytes can be data serialized by other languages.
print(fury.deserialize(data))
```

### What did you expect to see?

I expected the object to be serialized, deserialized, and printed.

### What did you see instead?

I got the error message below

```
Traceback (most recent call last):
  File "/tmp/x.py", line 10, in <module>
    fury.register_class(SomeClass, "example.SomeClass")
TypeError: Argument 'class_id' has incorrect type (expected int, got str)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by chaokunyang** [comment]
Hi @michaelmior , thanks for creating this issue. I changed `register_class ` to `register_class_tag`, and it can run successfully. 

The current API may be a little confusing:
```python
    def register_class(self, cls, class_id: int = None):
        self.class_resolver.register_class(cls, class_id=class_id)

    def register_class_tag(self, cls: type, type_tag: str = None):
        self.class_resolver.register_class_tag(cls, type_tag)
```

Maybe we should merge those methods into one method:
```python
    def register_class(self, cls, *, class_id: int = None, type_tag: str = None):
        if type_tag is not None:
           assert class_id is None
           self.class_resolver.register_class_tag(cls, type_tag=type_tag)
        else:
           self.class_resolver.register_class(cls, class_id=type_tag)
```

**Comment by michaelmior** [comment]
Thanks! That did resolve the issue for me. It seems like all that needs to be done for now then is update the example in the README.

**Comment by chaokunyang** [comment]
I will fix this in #1219, and release a new version later. We have some fix for java waiting being released too.
